File: striver/S2L2/ri.java
import java.util.Scanner;

public class ri {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for(int i=0;i<n;i++)
            arr[i] = sc.nextInt();
        ris(arr,1);
        for(int i=0;i<n;i++)
            System.out.print(arr[i]+" ");
    }
    public static void ris(int[] arr,int i) {
        if(i == arr.length)return;
        int val = arr[i];
        int j = i;
        while(j>=1 && arr[j-1]>=val){
            int temp = arr[j];
            arr[j] = arr[j-1];
            arr[j-1] = temp;
            j--;
        }
        ris(arr,i+1);
    }
}

File: striver/S2L2/ri.py
size = int(input())
arr = [0]*size
for i in range(0,size):
    arr[i]=int(input())
def rinsertion_sort(arr,i):
    if i == len(arr):
        return
    j=i
    while j>=1 and arr[j-1]>=arr[j]:
        temp = arr[j]
        arr[j] = arr[j-1]
        arr[j-1] = temp
        j-=1
    rinsertion_sort(arr,i+1)
rinsertion_sort(arr,1)
for i in range(0,size):
        print(arr[i],end=" ")